AVInput/AVInput.cpp

```diff
@@ -123,21 +169,87 @@ void AVInput::DeinitializeIARM()
         IARM_CHECK(IARM_Bus_UnRegisterEventHandler(
             IARM_BUS_DSMGR_NAME,
             IARM_BUS_DSMGR_EVENT_HDMI_IN_HOTPLUG));
+        IARM_CHECK(IARM_Bus_UnRegisterEventHandler(
+            IARM_BUS_DSMGR_NAME,
+            IARM_BUS_DSMGR_EVENT_HDMI_IN_SIGNAL_STATUS));
+        IARM_CHECK(IARM_Bus_UnRegisterEventHandler(
+            IARM_BUS_DSMGR_NAME,
+            IARM_BUS_DSMGR_EVENT_HDMI_IN_STATUS));
+        IARM_CHECK(IARM_Bus_UnRegisterEventHandler(
+            IARM_BUS_DSMGR_NAME,
+            IARM_BUS_DSMGR_EVENT_HDMI_IN_VIDEO_MODE_UPDATE));
+        IARM_CHECK(IARM_Bus_UnRegisterEventHandler(
+            IARM_BUS_DSMGR_NAME,
+            IARM_BUS_DSMGR_EVENT_HDMI_IN_ALLM_STATUS));
+        IARM_CHECK(IARM_Bus_UnRegisterEventHandler(
+            IARM_BUS_DSMGR_NAME,
+            IARM_BUS_DSMGR_EVENT_COMPOSITE_IN_HOTPLUG));
+        IARM_CHECK(IARM_Bus_UnRegisterEventHandler(
+            IARM_BUS_DSMGR_NAME,
+            IARM_BUS_DSMGR_EVENT_COMPOSITE_IN_SIGNAL_STATUS));
+        IARM_CHECK(IARM_Bus_UnRegisterEventHandler(
+            IARM_BUS_DSMGR_NAME,
+            IARM_BUS_DSMGR_EVENT_COMPOSITE_IN_STATUS));
     }
 }
 
 void AVInput::RegisterAll()
 {
-    Register<JsonObject, JsonObject>(_T("numberOfInputs"), &AVInput::endpoint_numberOfInputs, this);
-    Register<JsonObject, JsonObject>(_T("currentVideoMode"), &AVInput::endpoint_currentVideoMode, this);
-    Register<JsonObject, JsonObject>(_T("contentProtected"), &AVInput::endpoint_contentProtected, this);
+    CreateHandler({2});
+    Register<JsonObject, JsonObject>(_T(AVINPUT_METHOD_NUMBER_OF_INPUTS), &AVInput::endpoint_numberOfInputs, this);
+    Register<JsonObject, JsonObject>(_T(AVINPUT_METHOD_CURRENT_VIDEO_MODE), &AVInput::endpoint_currentVideoMode, this);
+    Register<JsonObject, JsonObject>(_T(AVINPUT_METHOD_CONTENT_PROTECTED), &AVInput::endpoint_contentProtected, this);
+    Register(AVINPUT_METHOD_GET_INPUT_DEVICES, &AVInput::getInputDevicesWrapper, this);
+    GetHandler(2)->Register<JsonObject, JsonObject>(AVINPUT_METHOD_GET_INPUT_DEVICES, &AVInput::getInputDevicesWrapper, this);
```

Comment:
what's GetHandler(2) for?

Reply:
Bringing changes from HDMIInput plugin. Inside HdmiInput Plugin, methods are registered with GenHandler(2).
Could you please confirm, do we need to follow the other methods registration present inside AVInput

Reply:
please don't use GenHandler(2) anywhere

Reply:
Removed GetHandler(2)